core/src/model/rewards/RewardSystem.java

```diff
@@ -0,0 +1,60 @@
+package model.rewards;
+
+import model.IPlayerCharacter;
+import model.playerperks.GlassCannon;
+import model.playerperks.SpeedDevil;
+import model.playerperks.Tank;
+
+public class RewardSystem {
+private boolean perkApplied;
+    //TODO fråga:  ska jag göra om till lista istället för ENUM? för att få bort getRandomRewards stora if-sats
+
+    public RewardSystem() {
+        initialize();
+    }
+
+    public void initialize() {
+        perkApplied = false;
```

Comment:
Why not set perkApplied to false directly when it is declared?

Reply:
iirc its best pratice to do it in a a initializer.